CLDR-14737 Update adding-locales.md (#4819)
unicode-org/cldr: docs/site/development/adding-locales.md

See [PR#4651](https://github.com/unicode-org/cldr/pull/4651).
- Check for possible clashes with similarly named files in [exemplars/main](https://github.com/unicode-org/cldr/tree/main/exemplars/main).
(For example, having an exemplars/main/xyz.xml file when common/main/xyz.xml is added will cause an error.)
If such a file is found, remove it (after checking whether any information in the file is useful).

#### Supply metadata
- In **common/supplemental/attributeValueValidity.xml**, add the language subtag to
	- `<variable id\="$language" type\="choice">` for TC locales, or
	- `<variable id\="$languageNonTcLtBasic" type\="choice">` for non-TC locales
- In **common/supplemental/supplementalMetadata.xml**, add the appropriate default content locale to \<defaultContent locales\="..."\>
	- The default content locale will include language and region (and possibly script, for multi-script languages).
- Consider whether the locale needs parent locale information (which it probably won't)
	- If parent inheritance can be determined by truncating subtags (and then falling back to root), no parent locale information is needed.
In the majority of cases this will be true.

> 	- If the locale is written in the non-preferred script for the language, it needs to have the parent specified as root,

Reviewer: ```suggestion
	- If the locale is written in the non-default script for the language, it needs to have the parent specified as root,
```

in contrast to `"CA"` which is in the `"regions"` attribute of an `"hours"` element where `"preferred"` attribute is `"h"`.
This corresponds to Canadian French (fr_CA) using 24-hour time even though 12-hour time is preferred for Canada (region CA).
	- For the locale submission for `"pap"` (ticket [CLDR-14872](https://unicode-org.atlassian.net/browse/CLDR-14872)
and pull request [PR#2542](https://github.com/unicode-org/cldr/pull/2542)),
the region `"CW"` did not appear in the `"timeData"` information and needed to be added to the appropriate location.

#### Optional additions
- Plural rules are not needed until the locale needs to move beyond Basic level,
but it is **strongly recommended** that plural rule information be collected as early as possible.
   - The addition of new plural forms to a language after data collection at higher levels has begun
will cause the coverage percentage to drop significantly until the data for plurals can be provided.
The addition of new plural forms can also be disruptive for any localized applications which are relying on CLDR plural data.

>    - If plural rule information is available, it can be added at the same time the locale is established.

Reviewer: ```suggestion
   - If plural rule information is available, it can be added at the same time the locale is established.
   - The simplest case is when the rules are identical to another locale's: it then just needs to be added to the list of locales with those rules.
```

Add the rules to **common/supplemental/plurals.xml** and **common/supplemental/ordinals.xml**.
- Similarly day period rules can be added to **common/supplemental/dayPeriods.xml**.

## Submit the pull request

- If requested, add to vendor targets in **tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/Locales.txt**,

> and to Cldr where resources are committed.

Reviewer: ```suggestion
and to CLDR where resources are committed.
```

Author: "Cldr" was in the original text. I think it refers to the entry in Locales.txt (where it is "Cldr" rather than "CLDR"). I'll go ahead and include this change. It seems pretty clear either way.

NOTE: this is an internal page for TC developers. Others should see [Core Data for New Locales](/index/cldr-spec/core-data-for-new-locales).

> ## Country locales
> 
> If you are just adding a new country locale (that is, CLDR already has a locale for the language), you just need to add the new empty locale file, and Update Survey Tool. 
> [This example](https://github.com/unicode-org/cldr/pull/4335/files) adds Kurdish in Syria ("ku-SY"). 

Reviewer 2: ```suggestion
## Regional locales

If you are just adding a new region locale (that is, CLDR already has a locale for the language), you just need to add the new empty locale file, and Update Survey Tool. 
[This example](https://github.com/unicode-org/cldr/pull/4335/files) adds Kurdish in Syria ("ku-SY"). 
```

- Verify or add time cycle information in **common/supplemental/supplementalData.xml**
	- In the majority of cases no changes will be needed.
	- In general terms, check that, under the \<timeData\> element, the \<hours\> element that has an attribute "regions" that contains the region for this locale, also has a "preferred" attribute that corresponds to the time cycle selected for the locale.
	- NB: Even though the attribute is "regions", it can contain a region or a locale. Consider the example of "fr_CA" in the "regions" attribute of an "hours" element where the "preferred" attribute is "H", in contrast to "CA" which is in the "regions" attribute of an "hours" element where "preferred" attribute is "h". This corresponds to Canadian French (fr_CA) using 24-hour time even though 12-hour time is preferred for Canada (region CA).
	- For the locale submission for "pap" (ticket [CLDR-14872](https://unicode-org.atlassian.net/browse/CLDR-14872) and pull request [PR#2542](https://github.com/unicode-org/cldr/pull/2542)), the region "CW" did not appear in the "timeData" information and needed to be added to the appropriate location.

#### Optional additions

> - Plural rules are not needed until the locale needs to move beyond Basic level. If plural rule information is available, it could be added at the same time the locale is established. Add them to **common/supplemental/plurals.xml** and **common/supplemental/ordinals.xml**.

Reviewer 3: Can we actually put that Plural rules are strong recommended? We've had several locales which have started contributing beyond basic without plural rules recently.

We may want to include what the impact is if plural rules are not available.

NOTE: this is an internal page for TC developers. Others should see [Core Data for New Locales](/index/cldr-spec/core-data-for-new-locales).

## Regional locales

If you are just adding a new region locale (that is, CLDR already has a locale for the language), you just need to add the new empty locale file, and Update Survey Tool. 
[This example](https://github.com/unicode-org/cldr/pull/4335/files) adds Kurdish in Syria ("ku-SY"). 

Otherwise:

## New locales

Before adding a new locale, you must have the minimal core data that cannot be added through the Survey Tool. See [Core Data for New Locales](/index/cldr-spec/core-data-for-new-locales) for the process.
This will include:
	- language code
	- exemplar sets (main, auxiliary, numbers, punctuation)
	- country / population data
	- time cycle (12-hour vs 24-hour, 0-based vs 1-based)
	- orientation of characters on a line and lines on a page
	- English name for this language (which won't be used until the language reaches Basic level, but needs to be documented in the CLDR ticket which adds the locale)

### Add Data to the GitHub repository

#### Verify or update country/population/language/script Data

> See [Update Language Script Info](/development/updating-codes/update-language-script-info) and [LikelySubtags and Default Content](/development/updating-codes/likelysubtags-and-default-content) for information on how to use the language, script, region and population data to verify that the information in [country_language_population.tsv](https://github.com/unicode-org/cldr/blob/main/tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/country_language_population.tsv) and [language_script.tsv](https://github.com/unicode-org/cldr/blob/main/tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/language_script.tsv) is correct. If changes are needed, correct an existing entry or add a new one as appropriate. Include references. 

Reviewer 3: If you want to enforce them, we either need a lint or a tool that automatically processes the markdown otherwise it is too hard to ensure consistency.
